crawl-ref/source/mon-cast.cc

```diff
@@ -5189,10 +5189,12 @@ static void _cast_resonance_strike(monster &caster, mon_spell_slot, bolt&)
         return;
 
     const int constructs = _count_nearby_constructs(caster, target->pos());
-    // base damage 3d(spell hd) (probably 3d12)
+    // base damage 3d(spell hd)
     // + 1 die for every 2 adjacent constructs (so at 4 constructs, 5dhd)
-    dice_def dice = resonance_strike_base_damage(caster);
+    const int pow = caster.spell_hd(SPELL_RESONANCE_STRIKE);
+    dice_def dice = resonance_strike_base_damage(pow);
     dice.num += div_rand_round(constructs, 2);
+    dice.num = min(MAX_RESONANCE_STRIKE_DAMAGE_DICE, dice.num);
```

Comment:
I'm a bit confused by this cap. If the target is surrounded by a full 8 constructs, they'd take (3 + 8 / 2 = ) 7 dice of damage. But you set `MAX_RESONANCE_STRIKE_DAMAGE_DICE = 6`. What's the point of that? If you're going to cap it, wouldn't it make more sense to cap it earlier, at 5 dice? This cap seems like it mostly does nothing, which seems like a bad use of complexity.